File: backend/app/api/routes/roster.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_db
from app.schemas.roster import DriverRosterItem, VehicleRosterItem
from app.services import roster_service

router = APIRouter(prefix="/roster", tags=["roster"])


@router.get("/drivers", response_model=list[DriverRosterItem])
def list_driver_roster(db: Session = Depends(get_db)):
    return roster_service.get_driver_roster(db)


@router.get("/vehicles", response_model=list[VehicleRosterItem])
def list_vehicle_roster(db: Session = Depends(get_db)):
    return roster_service.get_vehicle_roster(db)

File: frontend/src/api/roster.ts
import { apiClient } from "./client";
import type { DriverRosterItem, VehicleRosterItem } from "@/types/roster";

export async function fetchDriverRoster(): Promise<DriverRosterItem[]> {
  const { data } = await apiClient.get<DriverRosterItem[]>("/roster/drivers");
  return data;
}

export async function fetchVehicleRoster(): Promise<VehicleRosterItem[]> {
  const { data } = await apiClient.get<VehicleRosterItem[]>("/roster/vehicles");
  return data;
}
